File: java/completed/euler0007.java
import lib.EulerTime;

import java.util.ArrayList;
import java.util.Arrays;

/**
 * Created by Holden on 11/20/2016. - COMPLETED
 *
 * Problem 7 - 10001st Prime
 *
 */

public class Prob0007 {
    public static void main(String[] args) {
        long startTime = System.nanoTime();
        int answer = nthPrime(10001);
        long endTime = System.nanoTime();

        System.out.println(answer);

        long duration = (endTime - startTime);
        EulerTime.time(duration);
    }

    public static int nthPrime(int n) {
        ArrayList<Integer> primesArray = new ArrayList<>(Arrays.asList(3));
        int i = 3;
        int length = 2;
        boolean isPrime = true;

        while (length < n) {
            i += 2;

            for (int prime: primesArray) {
                if (i % prime == 0) {
                    isPrime = false;
                    break;
                }
            }

            if (isPrime) {
                primesArray.add(i);
                length++;
            } else {
                isPrime = true;
            }
        }

        return primesArray.get(n-2);
    }
}

File: python/completed/euler0007.py
###
#
# Created by Holden on 12/17/2015
#
# SOLVED on 12/17/2015
#
# Problem 7 - 10,001st Prime
#
###

def find_prime(number):
    prime_list = [2, 3]
    i = 3
    term_is_prime = True
    
    while len(prime_list) < number:
        i += 2
        
        for term in prime_list:
            if i % term == 0:
                term_is_prime = False
                break
                
        if term_is_prime:
            prime_list.append(i)
        else:
            term_is_prime = True
            
    return prime_list[-1]